### PR TITLE
Test simplifications

### PR DESCRIPTION
**Pull request by gssbzn**
This just looks at standardizing our use of `deep.Equal` instead of reflects methods
Also changes some errors to fatal to prevent future execution

### CODE REVIEW

File: cloudmanager/automation_config.go [hunk 1]
```diff
@@ -105,17 +105,17 @@ type SSL struct {
...
 	AutoPwd                  string                   `json:"autoPwd,omitempty"`
 	DeploymentAuthMechanisms []string                 `json:"deploymentAuthMechanisms"`
-	Disabled                 bool                     `json:"disabled"`
 	Key                      string                   `json:"key,omitempty"`
 	Keyfile                  string                   `json:"keyfile,omitempty"`
 	KeyfileWindows           string                   `json:"keyfileWindows,omitempty"`
 	UsersDeleted             []interface{}            `json:"usersDeleted"`
 	UsersWanted              []map[string]interface{} `json:"usersWanted"`
+	AuthoritativeSet         bool                     `json:"authoritativeSet"`
+	Disabled                 bool                     `json:"disabled"`
```
Comment (savannaendicott) [inline cloudmanager/automation_config.go:118]: Were these meant to be re-arranged? It seems they were previously in alphabetical order

File: cloudmanager/automation_config.go [hunk 2]
```diff
@@ -194,15 +194,15 @@ type LogRotate struct {
...
 	FeatureCompatibilityVersion string     `json:"featureCompatibilityVersion,omitempty"`
 	Hostname                    string     `json:"hostname,omitempty"`
-	LastGoalVersionAchieved     int        `json:"lastGoalVersionAchieved,omitempty"`
 	LogRotate                   *LogRotate `json:"logRotate,omitempty"`
-	ManualMode                  bool       `json:"manualMode,omitempty"`
 	Plan                        []string   `json:"plan,omitempty"`
 	ProcessType                 string     `json:"processType,omitempty"`
 	Version                     string     `json:"version,omitempty"`
+	Disabled                    bool       `json:"disabled,omitempty"`
+	ManualMode                  bool       `json:"manualMode,omitempty"`
```
Comment (savannaendicott) [inline cloudmanager/automation_config.go:207]: same comment as above, what is the reason for this change?

File: cloudmanager/projects.go [hunk 3]
```diff
@@ -28,7 +28,7 @@ const (
 
 // ProjectsService is an interface for interfacing with the Projects
 // endpoints of the MongoDB Atlas API.
-// See more: https://docs.atlas.mongodb.com/reference/api/projects/
+// See more: https://docs.cloudmanager.mongodb.com/reference/api/groups/
```
Comment (savannaendicott) [inline cloudmanager/projects.go:31]: This interface appears to be using the atlas api?

File: cloudmanager/projects.go [hunk 4]
```diff
@@ -37,8 +37,8 @@ type ProjectsService interface {
 	Delete(context.Context, string) (*atlas.Response, error)
 }
 
-//ProjectsServiceOp handles communication with the Projects related methos of the
-//MongoDB Atlas API
+// ProjectsServiceOp handles communication with the Projects related methos of the
```
Comment (savannaendicott) [inline cloudmanager/projects.go]: np: methos -> methods

File: cloudmanager/projects.go [hunk 5]
```diff
@@ -37,8 +37,8 @@ type ProjectsService interface {
...
+// MongoDB Cloud Manager API
```
Comment (savannaendicott) [inline cloudmanager/projects.go:41]: np: May have been the intention to add `See more: https://docs.cloudmanager.mongodb.com/reference/api/groups/` here as opposed to replacing it above? If not, maybe we could add that here anyway

File: cloudmanager/projects.go [hunk 6]
```diff
@@ -78,8 +78,8 @@ type Projects struct {
 	TotalCount int           `json:"totalCount"`
 }
 
-//GetAllProjects gets all project.
-//See more: https://docs.cloudmanager.mongodb.com/reference/api/groups/get-all-groups-for-current-user/
+// GetAllProjects gets all project.
```
Comment (savannaendicott) [inline cloudmanager/projects.go]: np: project -> projects

**Review by savannaendicott — APPROVED** [review]
a couple of nps/qs for typos & copy/pasting but overall LGTM

File: cloudmanager/projects.go (hunk 3 above)
Comment (gssbzn) [reply cloudmanager/projects.go:31]: what do you mean by this? the link was a copy/paste from the atlas client that I'm correcting now

File: cloudmanager/projects.go (hunk 5 above)
Comment (gssbzn) [reply cloudmanager/projects.go:41]: I haven't added any see more, I corrected the one we used to have, as those were copy/paste errors from when this was first done

File: cloudmanager/automation_config.go (hunk 1 above)
Comment (gssbzn) [reply cloudmanager/automation_config.go:118]: Yes, due to how memory allocation works on go these save a couple of bytes, https://medium.com/@sebassegros/golang-dealing-with-maligned-structs-9b77bacf4b97

File: cloudmanager/projects.go (hunk 3 above)
Comment (gssbzn) [reply cloudmanager/projects.go:31]: Talked offline

Comment (savannaendicott) [reply cloudmanager/projects.go:31]: discussed offline, comment above needs to be fixed as well

Comment (gssbzn) [reply cloudmanager/projects.go:31]: Fixed
